app/utils/ATSErrorHandler.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import com.google.inject.Inject
+import play.api.Logging
+import play.api.http.Status.{BAD_REQUEST, INTERNAL_SERVER_ERROR, NOT_FOUND}
+import play.api.mvc.Result
+import play.api.mvc.Results.{BadGateway, BadRequest, InternalServerError, NotFound}
+import uk.gov.hmrc.http.UpstreamErrorResponse
+
+class ATSErrorHandler @Inject()() extends Logging {
```

Comment:
Something to think about, if you extend Logging here, all the errors will be from ATSErrorHandler, so you might find it difficult to find out where the log/bug actually originates from